src/tests/ftest/security/cont_create_acl.py

```diff
@@ -0,0 +1,91 @@
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+from cont_security_test_base import ContSecurityTestBase
+
+class CreateContainterACLTest(ContSecurityTestBase):
+    # pylint: disable=too-few-public-methods,too-many-ancestors
+    """Tests container basics including create, destroy, open, query and close.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a CreateContainterACLTest object."""
+        super(CreateContainterACLTest, self).__init__(*args, **kwargs)
+
+    def test_container_basics(self):
+        """Test basic container create/destroy/open/close/query.
+
+            1. Create a pool (dmg tool) with no acl file passed.
+            2. Create a container (daos tool) with no acl file pased.
+            3. Destroy the container.
+            4. Create a container (daos tool) with a valid acl file passed.
+            5. Destroy the container.
+            6. Create a container (daos tool) with an invalid acl file passed.
+            7. Destroy the pool.
+            8. Remove all files created
+
+        :avocado: tags=all,full_regression,security,container_acl,
+        :avocado: tags=cont_create_acl,cont_create_acl
+        """
+        ## Create a pool
+        self.log.info("===> Creating a pool with no ACL file passed")
+        self.create_pool_with_dmg()
+
+        ## Create a container with no ACL file passed
+        self.log.info("===> Creating a container with no ACL file passed")
+        self.create_container_with_daos()
+
+        ## Destroy the container
+        self.log.info("===> Destroying the container")
+        self.destroy_container_with_daos()
+
+        ## Create a valid ACL file
+        self.log.info("===> Generating a valid ACL file")
+        self.generate_acl_file("valid")
+
+        ## Create a container with a valid ACL file passed
+        self.log.info("===> Creating a container with an ACL file passed")
+        self.create_container_with_daos("valid")
+
+        ## Destroy the container
+        self.log.info("===> Destroying the container")
+        self.destroy_container_with_daos()
+
+        ## Create an invalid ACL file
+        self.log.info("===> Generating an invalid ACL file")
+        self.generate_acl_file("invalid")
+
+        ## Create a container with an invalid ACL file passed
+        self.log.info("===> Creating a container with invalid ACL file passed")
+        self.create_container_with_daos("invalid")
+
```

Comment:
It looks a bit weird to me that the result isn't checked here, but inside the create_container_with_daos method. I'd argue it makes it harder to understand the flow of the tests. Just a suggestion, but having a separate helper function that checks whether the container was created, called verify_container_not_created or something like that, would make it obvious to the reader.

Reply:
Made different code changes in functions, in order to accept parameters and return values, so the validation of the test itself is done in the cont_create_acl test itself.
Thank you for the suggestion, you are right, it makes it easy to follow the flow of the test.